### PR TITLE
fix(ci+make+submodules): six instruments that reported fine and were measuring nothing

### PR DESCRIPTION
**Pull request by POWERFULMOVES**
Five items from the 2026-08-19 B850 sitrep, plus a sixth that landing them uncovered. One PR, six commits — they are separable but small, and splitting costs more review time than it saves at this team size.

Every one is the same shape: **an instrument that was reporting fine and was not measuring anything.**

---

## 1. Two SHA pins point at commits that do not exist

`actions/upload-artifact@50769540...` — commented `# v8.0.0`, a version never released — and `docker/setup-qemu-action@49b3bc8e...`. The GitHub API returns **422 "no commit found"** for both.

An unresolvable pin fails the workflow during **setup**: before `runs-on` resolves, before any step logs, before any annotation. The run records `startup_failure`, indistinguishable from a runner hiccup.

| workflow | runs | successes |
|---|---|---|
| `branch-protection-ruleset-sync` | 1 | **0, ever** |
| `branch-protection-drift` | 4 | **0, ever** |
| `build-nats-workers` | 5 (since 2026-07-26) | **0, ever** |

What that hid:

- **The 21-repo submodule enrollment merged in #2607 has never been applied.** `pmoves_standard.json` describes 25 enrolled repos and nothing has ever read it.
- Branch-protection drift has never once been evaluated.
- The NATS worker images have never been built by CI.

Worth naming: **#2569** added *14 regression tests* to the branch-protection pair. Those tests are green and the workflows have never executed a single step — the tests exercise the publisher module, not the workflow, and nothing checked that the workflow could start.

**Fixes** (verified against the API): upload-artifact → `043fb46d...` (v7.0.1), which the other 10 workflows already use, so all 12 pins now agree. setup-qemu-action → `c7c53464...` (v3.7.0), honoring the `# v3` intent rather than jumping the major.

**Gate**: `pmoves/tools/action_pin_audit.py`, wired into `merge-decision`. Checks both halves — the SHA must resolve (ERROR), and a `# vX.Y.Z` comment must name a real tag (WARN; a comment trailing a moving major tag is legitimate and is not flagged).

On an unreachable API it **exits 3, not 0**. A verifier that reports pass when it could not take a measurement is the exact defect this gate exists to catch.

> 227 pins across 59 workflows resolve. Falsification-tested: restoring both fabricated pins makes the gate report all 3 sites and exit 1. 16 unit tests, network stubbed, including one asserting the unreachable path exits 3.

---

## 2. The alias-divergence check from #2605 never landed

The remote branch head of `docs/instrument-trust-grounding` is `b49641227`, but #2605's squash took a state without `secrets_hardening_audit.py` — 10 files landed and that one was not among them. Verified `NOT on main`.

Cherry-picked back. It still finds a real divergence on B850 on first run:

```
[WARN] alias divergence in env.shared: KIMI_CODING_API (len 72),
       MOONSHOT_API_KEY (len 51) are the same manifest entry 'kimi_coding_api'
       but hold DIFFERENT values.
```

Whichever the funnel resolves first wins, and TensorZero reads the legacy name. WARN not ERROR — erroring would strand exactly the nodes that have the problem. Lengths, never values.

---

## 3. …and its remediation advice was a fix that undoes itself

The WARN said "delete the other from env.shared, then run secrets-funnel". That does not hold: `secrets-funnel` runs `secrets-local-hydrate` second, and `hydrate()` overlays from `local.env` whenever the env.shared value is a *placeholder* — and the empty string is the first entry in `PLACEHOLDER_VALUES`. A hand-deleted key is indistinguishable from one never set, so the stale value is written straight back. The operator watches the clear succeed and the next funnel run silently reverses it.

#2598 shipped the lever that expresses this correctly — `--clear KEY` blanks the value **and** tombstones it in `secrets_cleared.yaml`, which `hydrate()` skips. The message now points there and says why the manual route is not equivalent. Message text only; no behaviour change.

---

## 4. Command-line `ARGS` leaked into every sub-make

`make -C pmoves chit-manifest-register ARGS='--check'` — the invocation that target's **own help text documents** — never reached the tool:

```
bootstrap_light_env.py: error: unrecognized arguments: --check
make[1]: *** [mk/preflight.mk:39: env-bootstrap-lite] Error 2
make: *** [mk/codex.mk:84: chit-manifest-register] Error 2
```

GNU Make propagates command-line assignments to sub-makes via `MAKEFLAGS`. Each of these targets opens by recursing into `env-bootstrap-lite`, which passes `$(ARGS)` to a script with an entirely different flag set — so the target died on its first line, before its own tool ran.

7 targets fixed: `chit-manifest-register`, `chit-manifest-sync`, `chit-manifest-check`, `flight-check`, `flight-check-retro`, `topology-chit-gate`, `topology-chit-gate-strict`.

The fix must be a literal `ARGS=` **on the recursive `$(MAKE)` line**. Setting `ARGS =` inside `preflight.mk` would not work — a command-line override outranks any makefile assignment, and only another command-line assignment outranks an inherited one.

With the leak gone, `--check` reaches its tool and reports a pending manifest entry **nobody could see while the leak was in place** — the host-side CHIT key registered in #2605 was never projected into the v2 manifest, because the register step that does the projecting is one of the targets this leak broke. Applying it writes a protected manifest; left as an operator step.

**Swept the class, not the symptom:** 35 recursive `$(MAKE)` calls land on an ARGS-consuming target. 34 are aggregate targets whose own recipes do not take ARGS — latent, not currently broken. The one remaining same-class site, `smoke-qwen-rerank → smoke-gpu`, is deliberate: both drive `tools/smoke_gpu.py` with the same flags, so it is **allowlisted, not silenced**.

Ratchet: `pmoves/tests/make/test_args_no_leak_to_submake.py` (4 tests) — hard-fails on any `env-bootstrap-lite` recursion missing `ARGS=`, asserts those call sites still exist so the guard cannot pass vacuously, and fails on any *new* ARGS-consuming parent forwarding to a different tool. Falsification-tested.

---

## 5. The gitlink gate skipped the two submodules with no branch pin

`validate_submodule_gitlinks.sh` exists to catch a gitlink that is not on its tracked branch. When a submodule declared no `branch`, it printed `warn … skipped` and continued — so the submodules with **no declared strategy were the ones the strategy gate never checked**, and the run still went green.

Two of 72, and one was drifted:

```
pmoves-keygen        gitlink 36ef04163 is AHEAD of 'master'
PMOVES-nats-server   identical to 'main' — clean, just undeclared
```

`36ef041` is the head of open PR `POWERFULMOVES/pmoves-keygen#1`, not a commit on master. Load-bearing, not cosmetic: `keygen_cards.py:56-60` shell-builds `./cmd/keygen-cli`, which exists **only on that branch** — so a fresh clone of main resolves a submodule state that exists only inside an unmerged PR. The pin also predates the panic fix on that PR (`4224124`), so **main currently ships the panicking binary**. Every PR this gate ran on reported `ok` for it.

- `.gitmodules` — declare `branch = master` / `branch = main`. **72/72 now declare a tracked branch.**
- Validator — a missing branch pin is now **FAIL**, not warn-and-skip. Safe to close now that zero are undeclared.

**Verified both directions:**
- `--all` vs origin/main: pmoves-keygen now FAILs with *"is AHEAD of 'master' — commits not merged to the tracked branch"*, the message it should have produced all along. PMOVES-nats-server: `ok`.
- **changed-mode** vs origin/main — how the PR gate actually invokes it — **PASSES**. Only gitlinks that *change* in a PR are compared; this PR touches `.gitmodules` only, so nothing merge-ready today is retroactively blocked.

---

## 6. Found while landing this — the merge gate discards a whole chunk's results

CI failed this PR with 3 `ModuleNotFoundError: No module named 'slowapi'` in a file it does not touch. Chasing it produced the largest instrument finding of the set, and my first two explanations were both wrong before the measurement settled it.

`pytest_ratchet` groups tests by nearest-ancestor `conftest.py`, then chunks at 20 files in discovery order. **Chunk `pmoves/tests [1]` exceeds the 120s group timeout on every run**, and a timed-out chunk produces no JUnit report — so `parse_junit` sees *nothing* from it. Passes and failures alike are discarded, not just the hang. The baseline records the silence; it cannot record what the silence contains. That is 20 test files whose results have never counted.

And because chunk membership is a function of the **whole** discovered file list, adding any test file anywhere in `pmoves/tests` reshuffles the boundaries and changes which failures are visible. This PR's two test files pushed `test_content_provenance_gate.py` from chunk [1] into chunk [2], and three failures that had never once been reported appeared as "new".

Reproduced in a clean venv built from `.github/requirements-tests.txt`: pre-existing, nothing to do with these commits. `slowapi` is imported by `content-provenance-gate/main.py` and by **no test** — the same blind spot that requirements file already documents for `python-multipart`. Installing it is necessary but not sufficient; the tests then fail 401 and need real work in that service.

Two corrections worth recording, since the audit this work descends from is about not trusting confident-and-wrong instruments: I first blamed chunk-boundary shift using a hand-written model of the grouping, which said the file's group was unchanged — the model was wrong because I had guessed at the function name. I then blamed a recent regression; git said the code landed in #1415, long ago. Only running the ratchet's own `group_files` settled it.

**What this PR does about it:**

- Adds `pmoves/tests/make/conftest.py`. Without it, the new ARGS ratchet fell into chunk [1] itself — **CI would have discarded its results.** A gate whose own results are thrown away is not a gate. It is now its own one-file group: `[25/28] pmoves/tests/make (1 files)`, reported.
- Records the mechanism and the three specific hidden failures in `_known_failures.yaml` as comments.

**What it deliberately does not do:** baseline those three. That would convert *"invisible because a chunk timed out"* into *"known broken"* — a stronger and less true claim — and would silently absorb whatever else is hiding in those 20 files. The anchor returns that file to chunk [1], so this PR neither hides a new failure nor unilaterally surfaces an unknown number of old ones.

The real fix — stop discarding a timed-out chunk's partial output, or size chunks so none times out — surfaces an unknown number of further failures at once and needs an owner. Written down, not guessed at.

---

## Recorded, not fixed

`--all` reports **11 further pre-existing gitlink FAILs on main**: PMOVES-BoTZ, PMOVES-Danger-infra, PMOVES-transcribe-and-fetch, PMOVES.YT, PMOVES-MAI-UI, PMOVES-Headscale, PMOVES-OpenRoom (diverged/ahead), plus PMOVES-jcodemunch-mcp, pmoves-hirag-mcp, PMOVES-Spark-VSS (declared branch 404s on the fork). Changed-mode does not see them because those gitlinks are not moving. Each needs a per-repo decision — re-pin, or correct the declared branch — so they are left for their owners rather than guessed at here.

Also still true and unchanged by this PR: the job literally named `merge-gate` at `merge-gate.yml:16-29` is vacuous (`echo "passed=true"`). It is not in `merge-decision.needs`, so it is inert rather than harmful, but it reads as a gate in the checks list.

---

## Test results

```
pmoves/tests/tools/test_action_pin_audit.py          16 passed
pmoves/tests/tools/test_chit_manifest_registry.py    12 passed
pmoves/tests/make/test_args_no_leak_to_submake.py     4 passed
pmoves/tests/tools/test_silent_handler_sweep.py      22 passed
                                                     -----------
                                                     54 passed
```

`make -C pmoves action-pin-audit` → PASS, 227 pins / 59 workflows.
CI `python-tests` ratchet: **28 groups, 124 findings, 124 baselined, 0 new** — and `[25/28] pmoves/tests/make (1 files)` confirms the new gate's own group reported rather than being discarded.
All merge-gate jobs green: `action-pin-validation`, `python-tests`, `hardening-validation`, `merge-decision`, `submodule-gitlink-gate`.
`make -C pmoves secrets-audit` → exit 0, 6 warnings (the new divergence WARN + 5 pre-existing `.example`-vs-runtime gaps).
Full-suite collection: 1843 tests; 5 collection errors, all pre-existing in files this PR does not touch.

The `pytest_ratchet.py` run that `merge-gate` performs exceeds 10 minutes locally and was left to CI rather than reported as passing unverified.

---

## Operator steps this cannot do

1. **Dispatch `branch-protection-ruleset-sync.yml`** once this merges — it can start for the first time, and #2607's 21-repo enrollment applies.
2. **Clear the blank host-side CHIT key** with `bootstrap_env.py --clear` (`secrets_cleared.yaml` is still `cleared_keys: []`, so this has never been done). Blocked here by the zero-access guard on that key name.
3. **Apply the pending manifest entry** — `make -C pmoves chit-manifest-register` now works, but writes a protected manifest.
4. **Decide KIMI vs MOONSHOT** — which value the existing receipts were signed with. SPARK's call, not inferable from the env.
5. **Merge `pmoves-keygen#1`, then re-pin the parent gitlink to the resulting `master` sha.** Until then main ships the panic; after the re-pin, the gate in item 5 enforces it.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/merge-gate.yml [hunk 1]
```diff
@@ -88,9 +111,11 @@ jobs:
           echo "  python-tests: ${{ needs.python-tests.result }}"
           echo "  docker-build-validation: ${{ needs.docker-build-validation.result }}"
           echo "  hardening-validation: ${{ needs.hardening-validation.result }}"
+          echo "  action-pin-validation: ${{ needs.action-pin-validation.result }}"
           if [[ "${{ needs.python-tests.result }}" == "failure" ]] || \
              [[ "${{ needs.docker-build-validation.result }}" == "failure" ]] || \
-             [[ "${{ needs.hardening-validation.result }}" == "failure" ]]; then
+             [[ "${{ needs.hardening-validation.result }}" == "failure" ]] || \
+             [[ "${{ needs.action-pin-validation.result }}" == "failure" ]]; then
```
[the remark this answers is not in the document]
Comment (POWERFULMOVES) [reply .github/workflows/merge-gate.yml]: C:/Users/russe/AppData/Local/Temp/claude/C--Users-russe-OneDrive-Documents-GitHub-PMOVES-AI/eb22c84f-ec1b-4e4a-9b90-10cde196c80e/scratchpad/g1.md
